src/ui-components/Address.tsx

```diff
@@ -56,21 +58,56 @@ const Address = ({ address, className, displayInline, disableIdenticon, extensio
 	const router = useRouter();
 	const [username, setUsername] = useState('');
 	const [kiltName, setKiltName] = useState('');
+	const[isAutoGeneratedUsername,setIsAutoGeneratedUsername] = useState(true);
 
 	const encoded_addr = address ? getEncodedAddress(address, network) || '' : '';
+	const FEATURE_RELEASE_DATE = dayjs('2023-06-12').toDate(); // Date from which web3 address is being used as username on web3 signup.
+	const manualUsernames25char: string[] = ['marcosdanieldasilvasantos','masterquincyleroyharrison','sasha.fleur.polkaassembly'];
 
-	const fetchUsername = async () => {
+	const fetchUsername = async (isOnclick:boolean) => {
 		const substrateAddress = getSubstrateAddress(address);
-		if (!username && substrateAddress) {
+
+		if (substrateAddress) {
 			const { data, error } = await nextApiClientFetch<IGetProfileWithAddressResponse>(`api/v1/auth/data/profileWithAddress?address=${substrateAddress}`);
 			if (error) {
 				console.error(error);
 				return;
 			}
-			if (data && data.username) {
-				setUsername(data.username);
-				router.push(`/user/${data.username}`);
+
+			if(isOnclick){
+				if (data && data.username) {
+					setUsername(data.username);
```

Comment:
`setUsername(data.username);` is being called 3 times and the condition is always truthy, can be changed to one call.

Reply:
fixed